### ISSUE TITLE
Unable to list AWS roles in any way

### ISSUE THREAD

**Issue by fernando-villalba** [hashicorp/vault#9823]
I have checked the documentation multiple times and the path where my aws auth is mounted on and I cannot find anything wrong with my setup but this is failing:

`vault list aws/roles/
No value found at aws/roles/`

```
curl -s --header "X-Vault-Token: $(cat ~/.vault-token)" --request LIST "${VAULT_AH_ADD}/v1/aws/roles" | jq
{
  "errors": [
    "no handler for route 'aws/roles/'"
  ]
}

```

Is this a bug or am I missing something obvious?

I am following the documentation for [API](https://www.vaultproject.io/api-docs/secret/aws#list-roles) and [CLI](https://github.com/hashicorp/vault/issues/1545) as described so I can't find what I am doing wrong at all.

It would be nice to be able to see roles in the UI as well, even if not edit them directly there, would be helpful for reference.

**Comment by calvn** [comment]
That path looks correct to me. Did you [create any roles](https://www.vaultproject.io/api/secret/aws#create-update-role) previous to this list operation? "Roles" in this case is a resource specific to Vault and this secrets engine, and not the IAM roles on your AWS account.

**Comment by fernando-villalba** [comment]
Yes I know, and yes I have, which is why it's weird it's not listing anything at all. The roles were created with Terraform like so:

```
resource "vault_aws_auth_backend_role" "role" {
  for_each = var.services_object
  backend                         = "aws"
  role                            = "${var.environment}-${each.value.domain_name}-${each.key}"
  auth_type                       = "iam"
  token_ttl                       = var.aws_auth_backend_token_ttl
  token_max_ttl                   = var.aws_auth_backend_token_max_ttl
  token_policies                  = ["${var.environment}-${each.value.domain_name}-${each.key}"]
  bound_iam_principal_arns        = each.value.task_role_arns
}
```

**Comment by fernando-villalba** [comment]
The reason why this doesn't work is that the command needs to be written like this:

`vault list auth/aws/roles`

But any [references](https://github.com/hashicorp/vault/issues/1545) I saw in  pull requests didn't write the path as such, which is why it was confusing.
